### PR TITLE
8272345: macos doesn't check `os::set_boot_path()` result

### PR DESCRIPTION
**Pull request by iignatev**
Hi all,

could you please review this small fix?

from JBS:
> we don't check `os::set_boot_path()` return value on macos, which leads to a crash/assert down the road in case we weren't able to set BCP. on all other OSes, we terminate JVM w/ "Failed setting boot class path." error message.

testing:
- [x] tier1
- [x] `runtime/cds/appcds/MoveJDKTest.java`

Thanks,
-- Igor

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272345](https://bugs.openjdk.java.net/browse/JDK-8272345): macos doesn't check `os::set_boot_path()` result

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 1d8d2947ed21f0b631c53a70d6257142251fd36e

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5094/head:pull/5094` \
`$ git checkout pull/5094`

Update a local copy of the PR: \
`$ git checkout pull/5094` \
`$ git pull https://git.openjdk.java.net/jdk pull/5094/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5094`

View PR using the GUI difftool: \
`$ git pr show -t 5094`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5094.diff">https://git.openjdk.java.net/jdk/pull/5094.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/runtime/cds/appcds/MoveJDKTest.java [hunk 1]
```diff
@@ -25,16 +25,16 @@
 /*
  * @test
  * @summary Test that CDS still works when the JDK is moved to a new directory
+ * @bug 8272345
  * @requires vm.cds
- * @requires os.family == "linux"
+ * @requires os.family == "linux" | os.family == "mac"
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/cds/appcds/MoveJDKTest.java:30]: The test probably works everywhere but Windows if the issue is only symbolic links. If problems were reported on AIX or generic BSD then we could address that then.

**Review by dholmes-ora — APPROVED** [review]
Hi Igor,

Functional fix looks good. One suggestion with the test.

Thanks,
David

File: test/hotspot/jtreg/runtime/cds/appcds/MoveJDKTest.java (hunk 1 above)
Comment (iignatev) [reply test/hotspot/jtreg/runtime/cds/appcds/MoveJDKTest.java]: agree, I've updated the test to be runnable on all OSes but Windows.

**Comment by iignatev** [comment]
Thanks, David.

/integrate

**Pull request closed without merging**